core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -596,6 +628,36 @@ private void triggerUpdate()
         }
     }
 
+    /**
+     * Adaptively adjust batch size to meet expected request size:
+     * If requestSize is not equal to expectedSize, this function will try to estimate and adjust the batch size proportionally based on
+     * current nums of splits and size of request.
+     */
+    private int adjustSplitBatchSize(List<SplitAssignment> splitAssignments, long requestSize, long expectedSize, int currentSplitBatchSize)
+    {
+        if ((requestSize > expectedSize && currentSplitBatchSize > guaranteedSplitsPerRequest) || (requestSize < expectedSize && currentSplitBatchSize < maxUnacknowledgedSplits)) {
```

Comment:
Why is the `(requestSize < expectedSize && currentSplitBatchSize < maxUnacknowledgedSplits)` condition necessary? `adjustSplitBatchSize` is only called when `taskUpdateRequestJson.length > maxRequestSize`, right?